array_test.go

```diff
@@ -842,234 +842,237 @@ func TestArray_InList(t *testing.T) {
+				wantNotContains: success,
+			},
+			{
+				name:            "passing in an interface",
+				value:           []interface{}{[]interface{}{123, "foo"}},
+				wantContains:    failure,
+				wantNotContains: success,
+			},
+		}
 
-		value.NotContains("FOO")
-		value.chain.assert(t, success)
-		value.chain.clear()
+		for _, tc := range cases {
+			t.Run(tc.name, func(t *testing.T) {
+				reporter := newMockReporter(t)
 
-		value.Contains([]interface{}{123, "foo"})
-		value.chain.assert(t, failure)
-		value.chain.clear()
+				NewArray(reporter, []interface{}{123, "foo"}).Contains(tc.value...).
+					chain.assert(t, tc.wantContains)
+				NewArray(reporter, []interface{}{123, "foo"}).NotContains(tc.value...).
+					chain.assert(t, tc.wantNotContains)
+			})
+		}
 
-		value.NotContains([]interface{}{123, "foo"})
-		value.chain.assert(t, success)
-		value.chain.clear()
 	})
 
 	t.Run("canonization", func(t *testing.T) {
 		type (
 			myInt int
 		)
 
+		cases := []struct {
+			name            string
+			value           []interface{}
+			wantContains    chainResult
+			wantNotContains chainResult
+		}{
+			{
+				name:            "myInt check",
+				value:           []interface{}{myInt(123)},
+				wantContains:    success,
+				wantNotContains: failure,
+			},
+			{
+				name:            "float to int check",
+				value:           []interface{}{456.0},
+				wantContains:    success,
+				wantNotContains: failure,
+			},
+		}
+		for _, tc := range cases {
+			t.Run(tc.name, func(t *testing.T) {
+				reporter := newMockReporter(t)
+
+				NewArray(reporter, []interface{}{123, 456}).Contains(tc.value...).
+					chain.assert(t, tc.wantContains)
+				NewArray(reporter, []interface{}{123, 456}).NotContains(tc.value...).
+					chain.assert(t, tc.wantNotContains)
+			})
+		}
 		reporter := newMockReporter(t)
+		assert.Equal(t, []interface{}{123.0, 456.0}, NewArray(reporter,
+			[]interface{}{123, 456}).Raw())
 
-		value := NewArray(reporter, []interface{}{123, 456})
-
-		assert.Equal(t, []interface{}{123.0, 456.0}, value.Raw())
-
-		value.Contains(myInt(123))
-		value.chain.assert(t, success)
-		value.chain.clear()
-
-		value.NotContains(myInt(123))
-		value.chain.assert(t, failure)
-		value.chain.clear()
-
-		value.Contains(456.0)
-		value.chain.assert(t, success)
-		value.chain.clear()
-
-		value.NotContains(456.0)
-		value.chain.assert(t, failure)
-		value.chain.clear()
 	})
 
 	t.Run("invalid argument", func(t *testing.T) {
 		reporter := newMockReporter(t)
 
-		value := NewArray(reporter, []interface{}{})
+		NewArray(reporter, []interface{}{}).Contains(func() {}).
+			chain.assert(t, failure)
 
-		value.Contains(func() {})
-		value.chain.assert(t, failure)
-		value.chain.clear()
-
-		value.NotContains(func() {})
-		value.chain.assert(t, failure)
-		value.chain.clear()
+		NewArray(reporter, []interface{}{}).NotContains(func() {}).
+			chain.assert(t, failure)
 	})
 }
 
 func TestArray_ContainsAll(t *testing.T) {
 	t.Run("basic", func(t *testing.T) {
```

Comment:
Could you please also add one case, where:

* we pass multiple elements to ContainsAll (e.g. 2)
* they don't have duplicates (e.g. "foo", "bar")
* they are a subset of the array (e.g. array is "foo", "bar", baz")

Because it seems that for multiple arguments we check only "corner cases" (when we pass array copy or duplicate), but not checking a "regular" case.

Reply:
Same for TestArray_ContainsAny